flutter-studio/testSrc/com/android/tools/idea/tests/gui/framework/FlutterGuiTestRule.java

```diff
@@ -0,0 +1,351 @@
+  }
+
+  public IdeaFrameFixture importProject(@NotNull String projectDirName) throws IOException {
+    VirtualFile toSelect = VfsUtil.findFileByIoFile(setUpProject(projectDirName), true);
+    ApplicationManager.getApplication().invokeAndWait(() -> {
+      /* TODO(messick): Open the project in a new window. */
+    });
+    waitForBackgroundTasks();
+    return ideFrame();
+  }
+
+  /**
+   * Sets up a project before using it in a UI test:
+   * <ul>
+   * <li>Makes a copy of the project in testData/guiTests/newProjects (deletes any existing copy of the project first.) This copy is
+   * the one the test will use.</li>
+   * <li>Creates a Gradle wrapper for the test project.</li>
+   * <li>Updates the version of the Android Gradle plug-in used by the project, if applicable</li>
+   * <li>Creates a local.properties file pointing to the Android SDK path specified by the system property (or environment variable)
+   * 'ANDROID_HOME'</li>
+   * <li>Copies over missing files to the .idea directory (if the project will be opened, instead of imported.)</li>
+   * <li>Deletes .idea directory, .iml files and build directories, if the project will be imported.</li>
+   * <p/>
+   * </ul>
+   *
+   * @param projectDirName the name of the project's root directory. Tests are located in testData/guiTests.
+   * @throws IOException if an unexpected I/O error occurs.
+   */
+  private File setUpProject(@NotNull String projectDirName) throws IOException {
+    File projectPath = copyProjectBeforeOpening(projectDirName);
+
+    updateLocalProperties(projectPath);
+    cleanUpProjectForImport(projectPath);
+    return projectPath;
+  }
+
+  public File copyProjectBeforeOpening(@NotNull String projectDirName) throws IOException {
+    File masterProjectPath = getMasterProjectDirPath(projectDirName);
+
+    File projectPath = getTestProjectDirPath(projectDirName);
+    if (projectPath.isDirectory()) {
+      FileUtilRt.delete(projectPath);
+    }
+    FileUtil.copyDir(masterProjectPath, projectPath);
+    System.out.println(String.format("Copied project '%1$s' to path '%2$s'", projectDirName, projectPath.getPath()));
+    return projectPath;
+  }
+
+  protected void updateLocalProperties(File projectPath) throws IOException {
+    LocalProperties localProperties = new LocalProperties(projectPath);
+    localProperties.setAndroidSdkPath(IdeSdks.getInstance().getAndroidSdkPath());
+    localProperties.save();
+  }
+
+  @NotNull
+  protected File getMasterProjectDirPath(@NotNull String projectDirName) {
+    return new File(GuiTests.getTestProjectsRootDirPath(), projectDirName);
+  }
+
+  @NotNull
+  protected File getTestProjectDirPath(@NotNull String projectDirName) {
+    return new File(GuiTests.getProjectCreationDirPath(), projectDirName);
+  }
+
+  public void cleanUpProjectForImport(@NotNull File projectPath) {
+    File dotIdeaFolderPath = new File(projectPath, Project.DIRECTORY_STORE_FOLDER);
```

Comment:
I imagine some of this code is tickling our lint settings (like, preferring `final`)?

Reply:
If we're making fewer changes to this file - to minimize diffs - that makes sense.

Reply:
Yeah, we may have some conflicts between minimum diffs and lints. That's one aspect I have not thought about yet. If you have a preference I'll be glad to follow it.

Reply:
Minimizing diff s makes sense to me. We can perhaps turn off specific lints for whole files.